lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/LARSExpansion.cxx

```diff
@@ -0,0 +1,429 @@
+		break;
+	      }
+	    // Second test: the cross validation error is larger than an absolute
+	    // maximum and is not decreasing
+	    if (crossValidationError > maximumError)
+	      {
+		LOGINFO(OSS() << "Cross-validation error=" << crossValidationError << " larger than " << maximumError);
+		break;
+	      }
+	  } // No update of the best cross-validation error
+	// Third test: the best cross-validation error is below a given absolute threshold
+	if (!(bestCrossValidationError >= errorThreshold))
+	  {
+	    LOGINFO(OSS() << "Best cross-validation error=" << bestCrossValidationError << " smaller than threshold=" << errorThreshold);
+	    break;
+	  }
+	// Update the residuals
+	const Matrix designMatrix(leastSquaresMethod.computeWeightedDesign());
+	Point c;
+	// Least Angle Regression Stepwise
+	if (isLARS_)
+	  {
+	  }
+	// or orthogonal matching pursuit
+	else
+	  {
+	    c = coefficients;
+	  }
+	residuals = rhs - designMatrix * c;
+      } // Loop over the basis functions
+    // Store the best selection for the current output index
+    for (UnsignedInteger j = 0; j < bestSelection.getSize(); ++j)
+      {
+	// Deal only with non-zero coefficients
+	const Scalar marginalAlpha_kj = bestCoefficients[j];
+	// To avoid -0.0
+	if (std::abs(marginalAlpha_kj) > smallCoefficient)
+	  {
+	    // Current index in the decomposition of the current marginal output
+	    const UnsignedInteger index = bestSelection[j];
+	    // If the current index is not in the map, create it
+	    if (coefficientsMap.find(index) == coefficientsMap.end()) coefficientsMap[index] = Point(outputDimension, 0.0);
+	    // Add the current scalar coefficient to the corresponding component of the vectorial coefficient
+	    coefficientsMap[index][outputIndex] = marginalAlpha_kj;
+	  } // Nonzero coefficient
+      } // Loop over the marginal indices
+  } // Loop over the output dimension
+ // At this point, the map contains all the associations (index, vector coefficient). It remains to present these data into the proper form and to build the associated partial basis
+  std::map<UnsignedInteger, Point>::iterator iter;
+  // Full set of indices
+  Indices I_k(0);
+  // Full set of vectorial coefficients
+  Sample alpha_k(0, outputDimension);
+  // Full set of partial basis functions.
+  FunctionCollection Psi_k(0);
+  for (iter = coefficientsMap.begin(); iter != coefficientsMap.end(); ++iter)
+  {
+    const UnsignedInteger i = iter->first;
+    const Point currentCoefficient(iter->second);
+    I_k.add(i);
+    alpha_k.add(currentCoefficient);
+    // We could reuse the function
+    Psi_k.add(functions[i]);
+  }
+  // Build the result
+  result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, transformation_,
+                                  inverseTransformation_, basis_, I_k, alpha_k, Psi_k);
+  result_.setIsLeastSquares(true);
+  result_.setInvolvesModelSelection(true);
```

Comment:
Please use `FunctionalChaosResult.setSelectionHistory()` to store the history of OMP selection.